pkgs/top-level/python-packages.nix

```diff
@@ -12910,6 +12910,10 @@ self: super: with self; {
     inherit (pkgs.llvmPackages) openmp;
   };
 
+  pythran-numpy_2 = self.pythran.override {
+    numpy = numpy_2;
+  };
```

Comment:
And here OTH:

```
$ nix why-depends --all --precise nixpkgs#python312Packages.{pythran,numpy}
/nix/store/2gb5f7wh6pp3hxhvy16s9khp97b152a8-python3.12-pythran-0.15.0
└───bin/.pythran-config-wrapped: …3.12/site-packages','/nix/store/jc6k07xdz70lfwbfiffz9m2wn6ag3699-python3.12-numpy-1.26.4/lib/pyt…
    bin/.pythran-wrapped: …3.12/site-packages','/nix/store/jc6k07xdz70lfwbfiffz9m2wn6ag3699-python3.12-numpy-1.26.4/lib/pyt…
    bin/pythran: …':'}.PATH=${PATH/':''/nix/store/jc6k07xdz70lfwbfiffz9m2wn6ag3699-python3.12-numpy-1.26.4/bin'':'…
    bin/pythran-config: …':'}.PATH=${PATH/':''/nix/store/jc6k07xdz70lfwbfiffz9m2wn6ag3699-python3.12-numpy-1.26.4/bin'':'…
    nix-support/propagated-build-inputs: …ython3.12-gast-0.5.3 /nix/store/jc6k07xdz70lfwbfiffz9m2wn6ag3699-python3.12-numpy-1.26.4 /nix/st…
    → /nix/store/jc6k07xdz70lfwbfiffz9m2wn6ag3699-python3.12-numpy-1.26.4
```

Meaning that we could just `remove-references-to` numpy in the `pythran-config` script, and not propagate `numpy`. We could even go as far as to disallow references to `numpy`. This would allow anyone to choose their `numpy` when working with `pythran`.